app/src/main/java/com/android/wildex/ui/post/PostDetailsScreen.kt

```diff
@@ -95,18 +141,245 @@ fun PostDetailsScreen(
+@Composable
+fun CommentSection(
+    modifier: Modifier = Modifier,
+    commentsUI: List<CommentWithAuthorUI> = emptyList(),
+    userId: Id = "",
+    userProfilePictureURL: URL = "",
+    onProfile: (Id) -> Unit = {},
+    postDetailsScreenViewModel: PostDetailsScreenViewModel
+) {
+  Box(
+      modifier =
+          Modifier.padding(8.dp)
+              .border(2.dp, MaterialTheme.colorScheme.primary, RoundedCornerShape(12.dp))
+              .clip(RoundedCornerShape(24.dp))) {
+        Column(modifier = modifier.fillMaxWidth()) {
+          LazyColumn(
+              modifier = Modifier.weight(1f).fillMaxWidth(),
+              verticalArrangement = Arrangement.spacedBy(0.dp)) {
+                items(commentsUI) { commentUI ->
+                  Comment(commentUI = commentUI, onProfile = onProfile)
+                }
+              }
+
+          CommentInput(
+              userId = userId,
+              userProfilePictureURL = userProfilePictureURL,
+              onProfile = onProfile,
+              postDetailsScreenViewModel = postDetailsScreenViewModel)
+        }
+      }
 }
+
+@Composable
+fun Comment(
+    commentUI: CommentWithAuthorUI,
+    onProfile: (Id) -> Unit = {},
+) {
+  Box(modifier = Modifier.border(1.dp, MaterialTheme.colorScheme.primary).padding(14.dp)) {
+    Row(modifier = Modifier.fillMaxWidth(), verticalAlignment = Alignment.Top) {
+      ClickableProfilePicture(
+          modifier =
+              Modifier.size(48.dp)
+                  .testTag(ProfileScreenTestTags.COMMENT_PROFILE_PICTURE + commentUI.authorId),
+          profileId = commentUI.authorId,
+          profilePictureURL = commentUI.authorProfilePictureUrl,
+          onProfile = onProfile)
+
+      Spacer(modifier = Modifier.width(12.dp))
+
+      Column(modifier = Modifier.weight(1f)) {
+        Row(verticalAlignment = Alignment.CenterVertically) {
+          Text(
+              text = commentUI.authorUserName,
+          )
+          Spacer(modifier = Modifier.width(8.dp))
+          Text(
+              text = commentUI.date,
+          )
+        }
+
+        Spacer(modifier = Modifier.height(4.dp))
+
+        Text(text = commentUI.text, style = MaterialTheme.typography.bodyMedium)
+      }
+    }
+  }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun CommentInput(
+    userId: Id = "",
+    userProfilePictureURL: URL = "",
+    onProfile: (Id) -> Unit = {},
+    postDetailsScreenViewModel: PostDetailsScreenViewModel
+) {
+  Box(
+      modifier =
+          Modifier.border(1.dp, MaterialTheme.colorScheme.primary, RoundedCornerShape(12.dp))
+              .padding(8.dp)) {
+        Row(
+            modifier = Modifier.fillMaxWidth().padding(vertical = 4.dp),
+            verticalAlignment = Alignment.CenterVertically) {
+              ClickableProfilePicture(
+                  modifier =
+                      Modifier.size(56.dp)
+                          .testTag(ProfileScreenTestTags.CURRENT_USER_PROFILE_PICTURE),
```

Comment:
Same here to be coherent with my previous comment.
```suggestion
                          .testTag(ProfileScreenTestTags.testTagForProfilePicture(Firebase.auth.currentUser.uid)),
```